drivers/pds/pdsbackup/backuputils.go

```diff
@@ -0,0 +1,235 @@
+	// Pre-req: Azure bucket should be created by name pds-automation-1
+	log.Info("Add Azure backup creadentials")
+	accountKey := backupClient.azureStorageClient.accountKey
+	accountName := backupClient.azureStorageClient.accountName
+	backupCred, err := backupClient.components.BackupCredential.CreateAzureBackupCredential(tenantId, name, accountKey, accountName)
+	if err != nil {
+
+		return nil, fmt.Errorf("Error in adding the backup credentials to PDS , Err: %v ", err)
+	}
+	log.Infof("Adding backup target {Name: %v} to PDS.", name)
+	backupTarget, err := backupClient.components.BackupTarget.CreateBackupTarget(tenantId, name, backupCred.GetId(), bucketName, "", "azure")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create Azure backup target, Err: %v ", err)
+	}
+	log.Infof("[Backup Target: %v]Syncing to target clusters", name)
+	return backupTarget, nil
+}
+
+// CreateGcpBackupCredsAndTarget create backup creds,bucket and target.
+func (backupClient *BackupClient) CreateGcpBackupCredsAndTarget(tenantId, name string) (*pds.ModelsBackupTarget, error) {
+	log.Info("Create google storage for adding GCP backup target to control plane.")
+	err := backupClient.gcpStorageClient.createBucket(bucketName)
+	if err != nil {
+		return nil, fmt.Errorf("Failed while creating bucket, Err: %v ", err)
+	}
+	log.Info("Add GCP backup credentials")
+	projectId := backupClient.gcpStorageClient.projectId
+	gcpJsonKey := os.Getenv("GCP_JSON_PATH")
+
+	log.Infof("GCP Json PATH - %v projectId-%v ", gcpJsonKey, projectId)
+	backupCred, err := backupClient.components.BackupCredential.CreateGoogleCredential(tenantId, name, projectId, gcpJsonKey)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error in adding the backup credentials to PDS , Err: %v ", err)
+	}
+	log.Infof("Adding backup target {Name: %v} to PDS.", name)
+	backupTarget, err := backupClient.components.BackupTarget.CreateBackupTarget(tenantId, name, backupCred.GetId(), bucketName, "", "google")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create google backup target, Err: %v ", err)
+	}
+	return backupTarget, nil
+
+}
+
+// DeleteAwsS3BackupCredsAndTarget delete backup creds,bucket and target.
+func (backupClient *BackupClient) DeleteAwsS3BackupCredsAndTarget(backupTargetId string) error {
+	log.Info("Removing S3 backup creadentials and target from PDS.")
+	backupTarget, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+	credId := backupTarget.GetBackupCredentialsId()
+	log.Infof("Deleting backup target {Name: %v} to PDS.", backupTarget.GetName())
+	_, err := backupClient.components.BackupTarget.DeleteBackupTarget(backupTargetId)
+	if err != nil {
+		return fmt.Errorf("Failed to delete AWS S3 backup target, Err: %v ", err)
+	}
+	waitErr := wait.Poll(bkpTimeInterval, bkpMaxtimeInterval, func() (bool, error) {
+		model, bkpErr := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
```

Comment:
if error is expected , please validate the error message with the expected error